src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -259,6 +260,132 @@ ss::future<checked<std::nullopt_t, file_committer::errc>> commit_icb_files(
     co_return std::nullopt;
 }
 
+/// A table_committer accumulates files to commit to an Iceberg table and
+/// manages the commit process.
+///
+/// A file_committer will usually create one or more table_committer-s per
+/// topic, and will call process_pending_entry for each pending entry in the
+/// topic's state. Once all pending entries have been processed, the
+/// file_committer must call commit to commit the files to the Iceberg table.
+class table_committer {
```

Comment:
nit: the "committer" name makes it easy to mistakenly think that this is meant to be long-running, considering the file committer is also nearby. How about `pending_table_commit` or something? IMO it feels clearer that it's a single commit's worth of accounting

Reply:
`table_commit_builder` was my other choice